presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestOptimizedPartitionedOutputOperatorQueries.java

```diff
@@ -0,0 +1,560 @@
+    {
+        testIpAddress("tpch_bucketed");
+    }
+
+    public void testIpAddress(String schema)
+    {
+        String sql = "WITH lineitem_ipaddress AS \n" +
+                "(\n" +
+                "SELECT\n" +
+                "    partkey,\n" +
+                "    CAST(\n" +
+                "        CONCAT(\n" +
+                "            CONCAT(\n" +
+                "                CONCAT(\n" +
+                "                    CONCAT(\n" +
+                "                        CONCAT(\n" +
+                "                            CONCAT(CAST((orderkey % 255) AS VARCHAR), '.'),\n" +
+                "                            CAST((partkey % 255) AS VARCHAR)\n" +
+                "                        ),\n" +
+                "                        '.'\n" +
+                "                    ),\n" +
+                "                    CAST(suppkey AS VARCHAR)\n" +
+                "                ),\n" +
+                "                '.'\n" +
+                "            ),\n" +
+                "            CAST(linenumber AS VARCHAR)\n" +
+                "        ) AS IPADDRESS\n" +
+                "    ) AS ip\n" +
+                "    FROM " + schema + ".lineitem\n" +
+                "    )\n" +
+                "SELECT\n" +
+                "    CHECKSUM(l.ip) \n" +
+                "FROM lineitem_ipaddress l,\n" +
+                "    " + schema + ".partsupp p\n" +
+                "WHERE\n" +
+                "    l.partkey = p.partkey";
+
+        Object expected = computeActual(noAriaSession(), sql).getOnlyValue();
+        Object result = computeActual(ariaSession(), sql).getOnlyValue();
+        assertEquals(result, expected);
+    }
+
+    @Test
+    public void testBigIntWithNulls()
+    {
+        testIpAddress("tpch");
+    }
+
+    @Test
+    public void testBigIntWithNullsBucketed()
+    {
+        testIpAddress("tpch_bucketed");
+    }
+
+    public void testBigIntWithNulls(String schema)
+    {
+        String sql = "WITH lineitem_nulls AS (\n" +
+                "    SELECT\n" +
+                "        IF (mod(orderkey + linenumber, 11) = 0, NULL, partkey) AS l_partkey\n" +
+                "    FROM " + schema + ".lineitem\n" +
+                ")\n" +
+                "SELECT\n" +
+                "    checksum(l.l_partkey)\n" +
+                "FROM lineitem_nulls l,\n" +
+                "    " + schema + ".partsupp p\n" +
+                "WHERE\n" +
+                "    l.l_partkey = p.partkey";
+
+        Object expected = computeActual(noAriaSession(), sql).getOnlyValue();
+        Object result = computeActual(ariaSession(), sql).getOnlyValue();
+        assertEquals(result, expected);
+    }
+
+    @Test
+    public void testVarchar()
+    {
+        testVarchar("tpch");
+    }
+
+    @Test
+    public void testVarcharBucketed()
+    {
+        testVarchar("tpch_bucketed");
+    }
+
+    private void testVarchar(String schema)
```

Comment:
H2 supports this query as is, hence, the test can be written as follows:

```
private void testVarchar(String schema)
    {
        assertQuery(withSchema(schema), "SELECT\n" +
                "    l.comment\n" +
                "    FROM lineitem l, partsupp p\n" +
                "WHERE\n" +
                "    l.partkey = p.partkey\n" +
                "    AND l.suppkey = p.suppkey\n" +
                "    AND p.availqty < 1000");
    }

    private Session withSchema(String schema)
    {
        return Session.builder(getQueryRunner().getDefaultSession())
                .setSchema(schema)
                .build();
    }
```

However, I'm wondering if this test is needed at all? Don't we already have tests that involve re-partitioning with varchar columns?

Same for testInteger, testSmallint, testBigint tests.